mesonbuild/scripts/depfixer.py

```diff
@@ -26,6 +26,9 @@
 DT_SONAME = 14
 DT_MIPS_RLD_MAP_REL = 1879048245
 
+# Global cache for tools
+INSTALL_NAME_TOOL = False
```

Comment:
Shouldn't this be `INSTALL_NAME_TOOL = shutil.which('install_name_tool')` so it would get resolved on import? This would get of the `if` branching lookup code below.

Reply:
In that case we will call `shutil.which` uselessly on non-mac platforms, which can take a while if your `PATH` is very large on Windows (which will happen eventually once you've installed enough dev tools, frameworks, SDKs, etc). 

F.ex., on my brand-new Windows machine with an NVMe drive and just 22 entries in `PATH`, 1000 `shutil.which` calls in a loop take 2.7s. However, on my old Windows machine with a Samsung Evo 950 SSD and 48 entries in `PATH`, the same takes 30 seconds! I expect it to be much slower on rotating drives.

That's 30ms per `shutil.which` (when the disk is fully idle), which is a pretty significant startup-time cost (and unnecessary I/O) in the no-op `meson install --only-changed` workflow (which is equivalent to the "one gstreamer plugin changed, and I just want to install that" workflow) for doing unnecessary work.

Reply:
This is completely reasonable. Please add it as a comment above the variable declaration, otherwise someone might accidentally "optimize" it away in the future.